code/modules/clothing/glasses/glasses.dm

```diff
@@ -56,34 +56,28 @@
 /obj/item/clothing/glasses/meson/examine(mob/user)
 	..()
 	var/turf/T = get_turf(src)
-	if(T && T.z == ZLEVEL_MINING && mode && picked_excuse)
+	if(T && T.z == ZLEVEL_MINING && !mesons_on && picked_excuse)
 		to_chat(user, "<span class='warning'>Due to [picked_excuse], these Meson Scanners will not be able to display terrain layouts in this area.</span>")
 
 /obj/item/clothing/glasses/meson/proc/toggle_mode(mob/user)
-	if(vision_flags & SEE_TURFS)
-		mode = TRUE
-		vision_flags &= ~SEE_TURFS
-		if(ishuman(user))
-			var/mob/living/carbon/human/H = user
-			if(picked_excuse)
-				to_chat(H, "<span class='warning'>Due to [picked_excuse], your Meson Scanners will not be able to display terrain layouts in this area.</span>")
-			if(H.glasses == src)
-				H.update_sight()
-	else if(!(vision_flags & SEE_TURFS))
-		mode = FALSE
-		vision_flags |= SEE_TURFS
-		if(ishuman(user))
-			var/mob/living/carbon/human/H = user
-			to_chat(H, "<span class='notice'>Your Meson Scanners have reactivated.</span>")
-			if(H.glasses == src)
-				H.update_sight()
+	vision_flags ^= SEE_TURFS
+	mesons_on = (vision_flags & SEE_TURFS)? TRUE : FALSE
+
+	if(iscarbon(user)) //only carbons can wear glasses

```

Comment:
So this entire section doesn't apply to borgs who have mesons, such as the engieborg? I see you've removed mesons from miner borgs.